rollingpin/main.py

```diff
@@ -120,11 +120,13 @@ def _parse_args(config, raw_args):
 def _select_hosts(config, args):
     # get the list of hosts from the host source
     try:
-        all_hosts = yield config["hostsource"].get_hosts()
+        all_hosts_unsorted = yield config["hostsource"].get_hosts()
     except HostSourceError as e:
         print_error("could not fetch host list: {}", e)
         sys.exit(1)
 
+    all_hosts = interleaved(all_hosts_unsorted, key=lambda h: h.pool)
```

Comment:
Cool!  Much nicer to have the interleaving logic here and let the other layer be unconcerned with it.